crate2nix/src/prefetch.rs

```diff
@@ -165,13 +198,14 @@ fn get_command_output(cmd: &str, args: &[&str]) -> Result<String, Error> {
         .map_err(|_e| format_err!("output of '{} {}' is not UTF8!", cmd, args.join(" ")))
 }
 
-trait PrefetchableSource: ToString {
+#[async_trait]
```

Comment:
I fear that bringing in async into this project - as is - has no advantage but adds a lot of dependencies.

We are still spawning a process for every fetch...

Reply:
Yes, but the process is completely IO bound - one place where async shines. I'm actually using the single-threaded scheduler in this implementation precisely because the computation is IO bound and there would be nothing beneficial about scheduling a bunch of threads that simply wait on IO.

I might be missing something though, I'm relatively new to Rust and to async for that matter.